sources/packages/web/src/views/aest/institution/DesignationAESTView.vue

```diff
@@ -0,0 +1,221 @@
+        location => ({
+          locationId: location.locationId,
+          locationName: location.locationName,
+          requestForDesignation: location.requested,
+          approvedForDesignation: location.approved,
+          locationAddress: formatter.getFormattedAddress({
+            ...location.locationData.address,
+            provinceState: location.locationData.address.province,
+          }),
+        }),
+      );
+    };
+    onMounted(async () => {
+      await loadDesignation();
+    });
+
+    const updateDesignation = async (
+      designationStatus: DesignationAgreementStatus,
+    ) => {
+      if (designationStatus === DesignationAgreementStatus.Approved) {
+        /*If the update action is Approval, the build the designation location array for form
+          by merging the institution locations list with designation locations list
+        */
+        const institutionLocations = await InstitutionService.shared.getAllInstitutionLocationSummary(
+          designationAgreement.value.institutionId,
+        );
+        updateDesignationModel.value.startDate =
+          designationAgreement.value.startDate;
+        updateDesignationModel.value.endDate =
+          designationAgreement.value.endDate;
+        updateDesignationModel.value.designationStatus = designationStatus;
+        updateDesignationModel.value.institutionId =
+          designationAgreement.value.institutionId;
+        updateDesignationModel.value.locationsDesignations = institutionLocations?.map(
+          location =>
+            ({
+              locationId: location.id,
+              locationName: location.name,
+              requested: false,
+              locationAddress: formatter.getFormattedAddress({
+                ...location.data.address,
+                provinceState: location.data.address.province,
+              }),
+            } as UpdateDesignationLocationDto),
+        );
+        updateDesignationModel.value.locationsDesignations.forEach(location => {
+          const requestedDesignation = designationAgreement.value.locationsDesignations.find(
+            designationLocation =>
+              designationLocation.locationId === location.locationId,
+          );
+          if (requestedDesignation) {
+            location.approved = requestedDesignation.approved !== false;
+            location.designationLocationId =
+              requestedDesignation.designationLocationId;
+            location.requested = requestedDesignation.requested;
+          }
+        });
+      }
+      const response = await approveDenyDesignationModal.value.showModal();
+      //Update designation only on a submit action.
+      if (response) {
+        try {
+          await DesignationAgreementService.shared.updateDesignationAgreement(
+            props.designationAgreementId,
+            response as UpdateDesignationDto,
+          );
+          await loadDesignation();
```

Comment:
Can we move the _await loadDesignation()_ under the toast?

Reply:
Moved.